packages/web3-providers/src/Web3/Base/state/Network.ts

```diff
@@ -117,7 +117,7 @@ export class NetworkState<ChainId, SchemaType, NetworkType>
     }
 
     async updateNetwork(id: string, updates: Partial<TransferableNetwork<ChainId, SchemaType, NetworkType>>) {
-        const network = this.assertNetwork(id)
+        const network = this.storage.networks.value[id]
```

Comment:
The network to be updated MUST exist. Object property accessor might return `undefined`.